### PR TITLE
General docs polish (Pool Hooks package)

### PR DESCRIPTION
**Pull request by EndymionJkb**
# Description

Mainly a docs update, it does also make some minor code changes, such as enforcing consistent naming, and adding missing tests (e.g., withdrawals from the hook fee pool).

## Type of change

- [ ] Bug fix 
- [ ] New feature 
- [ ] Breaking change 
- [ ] Dependency changes
- [X] Code refactor / cleanup
- [ ] Optimization: [ ] gas / [ ] bytecode
- [ ] Documentation or wording changes
- [ ] Other

## Checklist:

- [X] The diff is legible and has no extraneous changes
- [X] Complex code has been commented, including external interfaces
- [ ] Tests have 100% code coverage
- [X] The base branch is either `main`, or there's a description of how to merge

### CODE REVIEW

File: pkg/pool-hooks/contracts/FeeTakingHookExample.sol [hunk 1]
```diff
@@ -161,23 +177,34 @@ contract FeeTakingHookExample is BaseHooks, Ownable {
...
+    /**
+     * @notice Sets the hook remove liquidity fee percentage, charged on every remove liquidity operation.
+     * @dev This function must be permissioned.
+     */
+    function setRemoveLiquidityHookFeePercentage(uint64 hookFeePercentage) external onlyOwner {
         removeLiquidityHookFeePercentage = hookFeePercentage;
     }
+
+    /// @notice Withdraws the accumulated fees. This function must be permissioned.
+    function withdrawFees(IERC20 feeToken) external onlyOwner {
```
Comment (jubeira) [inline pkg/pool-hooks/contracts/FeeTakingHookExample.sol:207]: This doesn't strictly need to be permissioned if you're transferring the funds to the owner.

File: pkg/pool-hooks/test/foundry/ExitFeeHookExample.t.sol [hunk 2]
```diff
@@ -30,6 +28,9 @@ contract ExitFeeHookExampleTest is BaseVaultTest {
     uint256 internal daiIdx;
     uint256 internal usdcIdx;
 
+    // 10% exit fee
+    uint64 exitFeePercentage = 1e17;
```
Comment (jubeira) [inline pkg/pool-hooks/test/foundry/ExitFeeHookExample.t.sol:32]:
```suggestion
    uint64 exitFeePercentage = 10e16;
```

File: pkg/pool-hooks/test/foundry/VeBALFeeDiscountHookExample.t.sol [hunk 3]
```diff
@@ -116,44 +115,44 @@ contract VeBALFeeDiscountHookExampleTest is BaseVaultTest {
...
 
-        // Even if bob has veBAL, since he is using an untrusted router, he will get no discounts
+        // Even if Bob has veBAL, since he is using an untrusted router, he will get no discount.
         _doSwapAndCheckBalances(untrustedRouter);
     }
 
     function _doSwapAndCheckBalances(address payable routerToUse) private {
-        // 10% swap fee. Since vault does not have swap fee, the fee will stay in the pool
+        // 10% swap fee. Since the Vault has no swap fee, the fee will stay in the pool.
         uint256 swapFeePercentage = 1e17;
```
Comment (jubeira) [inline pkg/pool-hooks/test/foundry/VeBALFeeDiscountHookExample.t.sol:144]:
```suggestion
        uint256 swapFeePercentage = 10e16;
```

**Review by jubeira — APPROVED** [review]
LGTM, great job @EndymionJkb !

File: pkg/pool-hooks/contracts/ExitFeeHookExample.sol [hunk 4]
```diff
@@ -19,19 +19,44 @@ import {
...
-    /// @dev Exit hook fee above limit.
-    error ExitHookFeeAboveLimit(uint256 fee, uint256 limit);
-
-    /// @dev Pool does not support adding liquidity through donation.
+    // Percentages are represented as 18-decimal FP numbers, which have a maximum value of 1e18 (100%),
+    // so 60 bits are sufficient.
+    uint64 public exitFeePercentage;
+
+    // Maximum exit fee of 10%
+    uint64 public constant MAX_EXIT_FEE_PERCENTAGE = 1e17;
```
[the remark this answers is not in the document]
Comment (EndymionJkb) [reply pkg/pool-hooks/contracts/ExitFeeHookExample.sol:46]: Ok; we seem to be converging on showing percentages as 20e16, vs. 0.2e18 (or 2e17).

File: pkg/pool-hooks/contracts/FeeTakingHookExample.sol (hunk 1 above)
Comment (EndymionJkb) [reply pkg/pool-hooks/contracts/FeeTakingHookExample.sol:207]: Right; unless we had a recipient, which is overkill here.

**Pull request merged**